AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/SignedInData.swift

```diff
@@ -15,16 +15,14 @@ struct SignedInData {
     let deviceMetadata: DeviceMetadata
     let cognitoUserPoolTokens: AWSCognitoUserPoolTokens
 
-    init(
-        userId: String,
-        userName: String,
-        signedInDate: Date,
-        signInMethod: SignInMethod,
-        deviceMetadata: DeviceMetadata = .noData,
-        cognitoUserPoolTokens: AWSCognitoUserPoolTokens
+    init(signedInDate: Date,
+         signInMethod: SignInMethod,
+         deviceMetadata: DeviceMetadata = .noData,
+         cognitoUserPoolTokens: AWSCognitoUserPoolTokens
     ) {
-        self.userId = userId
-        self.userName = userName
+        let user = try? TokenParserHelper.getAuthUser(accessToken: cognitoUserPoolTokens.accessToken)
+        self.userId = user?.userId ?? "unknown"
+        self.userName = user?.username ?? "unknown"
```

Comment:
This will eventually be passed back to the customer.. Are we okay passing `unknown` value back?

Reply:
As discussed this will be always passed back from Cognito. "Unknown" is just a fallback so that we do not need to crash the app, in case parsing fails.